### DISCUSSION TITLE
Resovling Merge Conflicts by Multiple Teams

### DISCUSSION THREAD

**Discussion by erichjzimmerman** [community/community#56527] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

We have a large repository that accepts changes from many different teams.

I want to merge the `main` branch into `myfeature` branch.

```
> git checkout -b myfeature-merge main
> git merge myfeature
```

This results in a number of conflicts. The thing is, I want different teams to resolve the conflicts. It's not a problem to make sure conflict resolutions don't overlap (i.e. two different teams will not attempt to resolve the same conflicts).

I don't know how to divide up that responsibility. If I try to just fix some of the conflicts, Git will not let me commit the changes, because there are unmerged changes. However, if I only stage some files, reverting the unresolved conflicts, Git thinks the merge has been successful.

The only approach that seems *might* work is to:
* Capture the list of conflicting files
* Add and commit the conflict files (with the <<<< ==== >>>> lines in)
* Farm out the work to different teams until all the conflicting files are complete

But doing it this way, Gt is no longer aware of the conflicts, so it becomes a management thing on my end (searching files, making sure they are all complete).

What I would _like_ is some way to incrementally merge conflicts; so that when other teams do their work, the number of conflicts gradually reduces.

**Comment by jge162** [comment]
Hello @erichzimmermansonos You can create a new branch for incrementally managing conflicts.
- Create new branch
`git checkout -b myfeature-merge main`
- Next do the merge with myfeature 
`git merge myfeature`
- Then stage each conflict for each team member to resolve
`git add --patch "conflicting-file"`
- After everyone has reviewed conflicts commit the changes, and go from there. repeat `git merge` and `git add` as needed till all conflicts have been resolved
`git commit -m "Resolved conflicts in 'conflicting-file'`


perhaps consider running `git pull origin main` to help keep all branches up to date too.

**Reply by erichjzimmerman** [reply]
Thanks for the help!

At step 3, when I try `git add --patch "<file>"`, I end up with the following message:
```
ignoring unmerged: <file-name>
No changes.
```

And the Unmerged Paths stays the same.

**Reply by jge162** [reply]
Hello @erichzimmermansonos ignoring unmerged is telling you files you are trying to merge are unmerged. 

- Use `git status` to see the list of unmerged files and their status.
- step 2: Manually edit the conflicting file, resolving the conflicts.
- step 3: After resolving the conflicts, use `git add <file>` to stage the resolved file.
- Repeat steps 2 and 3 for each conflicting file until all conflicts are resolved.
- Once all conflicts are resolved and staged, use `git commit -m "Resolved conflicts in 'conflicting-file'"` to commit the changes.

**Reply by erichjzimmerman** [reply, marked as answer]
I aprpeciate the response.
That makes sense, but the original problem is that I want different team members to resolve their own conflicts.

(Admittedly, this is in large part because I wasn't vigilant on merging over a decently long period of time)

So I do the following:
```
git checkout -b myfeature-merge main
git merge myfeature
```
At this point, I see listed 20 - 30 conflicting files.

Ideally, developers A, B, and C will each resolve some number of those conflicts, from their own machine.

Everything I am seeing seems that I have to resolve all the conflicts after calling `git merge` before I can make other changes available to anyone else.

Is this a place where `git reset` or `git revert` comes in handy? Should each developer attempt the merge, resolve the conflicts for their files, but use `git reset` to ignore other conflicting files? And will those files still report merge conflicts to the other developers?

I think it might be pretty simple, and I'm just not getting it.

**Reply by jge162** [reply]
Yes that is correct using git reset to unstage specific files, developers can focus on resolving conflicts in their own set of files without affecting the changes made by others
